internal/audit/kafka/publisher.go

```diff
@@ -170,18 +168,12 @@ func (p *Publisher) WriteDecisionLogEntry(ctx context.Context, record audit.Deci
 
 func (p *Publisher) write(ctx context.Context, msg *kgo.Record) error {
 	if p.sync {
-		produceCtx, produceCancel := context.WithTimeout(ctx, p.produceTimeout)
-		defer produceCancel()
-
-		return p.Client.ProduceSync(produceCtx, msg).FirstErr()
+		return p.Client.ProduceSync(ctx, msg).FirstErr()
 	}
 
 	// detach the context from the caller so the request can return
 	// without cancelling any async kafka operations
-	produceCtx, produceCancel := context.WithTimeout(context.Background(), p.produceTimeout)
-	defer produceCancel()
-
-	p.Client.Produce(produceCtx, msg, func(r *kgo.Record, err error) {
+	p.Client.Produce(context.Background(), msg, func(r *kgo.Record, err error) {
```

Comment:
This is the actual fix, completely detaching the context cancellation from the parent.

The other changes in the PR cover the integration tests & dropping the produce timeout config, which is no longer required.